common/src/main/java/com/wynntils/core/features/overlays/OverlayManager.java

```diff
@@ -45,35 +55,143 @@ public OverlayManager(CrashReportManager crashReportManager) {
         addCrashCallbacks();
     }
 
+    // region Initialization and Registration
+
     public void registerOverlay(
-            Overlay overlay, RenderEvent.ElementType elementType, RenderState renderAt, Feature parent) {
-        overlayInfoMap.put(overlay, new OverlayRenderInfo(elementType, renderAt));
-        overlayParent.put(overlay, parent);
+            Overlay overlay,
+            Feature parent,
+            RenderEvent.ElementType elementType,
+            RenderState renderAt,
+            boolean enabledByDefault) {
+        overlayParentMap.putIfAbsent(parent, new LinkedList<>());
+        overlayParentMap.get(parent).add(overlay);
+
+        overlayInfoMap.put(overlay, new OverlayInfoContainer(parent, elementType, renderAt, enabledByDefault));
     }
 
     public void unregisterOverlay(Overlay overlay) {
+        overlayParentMap.get(overlayInfoMap.get(overlay).parent()).remove(overlay);
+
         overlayInfoMap.remove(overlay);
-        overlayParent.remove(overlay);
         enabledOverlays.remove(overlay);
     }
 
-    public void disableOverlays(List<Overlay> overlays) {
-        enabledOverlays.removeIf(overlays::contains);
-        overlays.forEach(
+    public void disableOverlays(Feature parent) {
+        overlayParentMap.getOrDefault(parent, Collections.emptyList()).forEach(this::disableOverlay);
```

Comment:
As a style issue, we have mostly used `List.of()` instead of `Collections.emptyList()`.